session.go

```diff
@@ -251,28 +260,26 @@ func (s *Session) Path() string {
 // Session message senders
 
 func (s *Session) sendClientSetup() error {
-	params := map[uint64]wire.Parameter{
-		wire.MaxSubscribeIDParameterKey: wire.VarintParameter{
-			Type:  wire.MaxSubscribeIDParameterKey,
-			Name:  "max_subscribe_id",
-			Value: s.MaxSubscribeID,
+	params := wire.Parameters{
+		wire.KeyValuePair{
+			Type:        wire.MaxRequestIDParameterKey,
+			ValueVarInt: s.localMaxRequestID.Load(),
 		},
 	}
 	if s.Protocol == ProtocolQUIC {
 		path := s.path
-		params[wire.PathParameterKey] = wire.StringParameter{
-			Type:  wire.PathParameterKey,
-			Name:  "path",
-			Value: path,
-		}
+		params = append(params, wire.KeyValuePair{
+			Type:       wire.PathParameterKey,
+			ValueBytes: []byte(path),
+		})
 	}
 	return s.ctrlMsgSendQueue.enqueue(context.Background(), &wire.ClientSetupMessage{
 		SupportedVersions: wire.SupportedVersions,
 		SetupParameters:   params,
 	})
 }
 
-// Subscribe subscribes to track in namespace using id as the subscribe ID. It
+// Subscribe subscribes to track in namespace using id as the request ID. It
```

Comment:
This comment sounds like id is a parameter. It would be better to tell that it generates a new requestID and trackAlias and use these.

I think that it would also be good to return both the requestID and the trackAlias for reference and possible future use when updating a subscription.

Reply:
I removed the whole point about request IDs from the comment.

Would it be enough to return them via the RemoteTrack as you also implemented in https://github.com/mengelbart/moqtransport/pull/204?

Reply:
Well, as far as I understand, the RequestID from the RemoteTrack is for the publisher, while the return value would be directly to the Subscriber who calls Subscribe. The subscriber need a handle so that it can make SUBSCRIBE_UPDATE later. My thought was that RequestID would be needed to call such an update function.
Maybe things become clearer once I try to implement the track switch.

Reply:
The `RemoteTrack` object is what a subscriber uses to interact with the incoming track (e.g., read from it or unsubscribe from it). So I think it should store the request ID of the subscribe message that opened the track.
As for subscribe update, I think it could be a nice API to implement an `update` function on the remote track, but not sure if that really makes sense or if it should rather be done on the session. I guess we can try one implementation, and if it turns out to be a bad API, we can still change it later.